### PR TITLE
[BUG] Rewrite Bigint SlotRef to compare DecimalLiteral in Binary predicate

### LINKED ISSUE #7264: [Bug] Bigint key column in where clause equal double constant, bucket cannot be hit
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

trunk

### What's Wrong?

When the bigint key column in the where clause is equal to the double constant, the bucket cannot be hit.

Reproduce on the ssb test set:
`select * from LINEORDER where LO_ORDERKEY = 33.0 limit 1;`

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Convert the binary predicate of the form
`<CastExpr<SlotRef(ResultType=BIGINT)>> <op><DecimalLiteral>` 
to the binary predicate of 
`<SlotRef(ResultType=BIGINT)> <new op> <new DecimalLiteral>`, 
thereby allowing the binary predicate The predicate pushes down and completes the bucket clipped.

For query `select * from T where t1 = 2.0`, when the ResultType of column t1 is equal to BIGINT, in the binary predicate analyze, the type will be unified to DECIMALV2, so the binary predicate will be converted to `<CastExpr<SlotRef>> <op In the form of ><DecimalLiteral>`, because Cast wraps the t1 column, it cannot be pushed down, resulting in poor performance.We convert it to the equivalent query `select * from T where t1 = 2` to push down and improve performance.

SSB test:
1. query `select * from LINEORDER3 where LO_ORDERKEY <2.2`
Performance improvement: `1.587s` -> `0.012s`,
The result and performance of `select * from LINEORDER3 where LO_ORDERKEY <3` are equivalent, and the other comparison methods are the same.
2. query `select * from LINEORDER3 where LO_ORDERKEY = 2.2`
Performance improvement: `0.012s` -> `0.006`.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #7264) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yangzhg** [comment]
We need more discuss

**Review by yangzhg — CHANGES_REQUESTED** [review]
We need more discuss

**Comment by xinyiZzz** [comment]
> We need more discuss

Modified according to discussion

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java [hunk 1]
```diff
@@ -231,6 +231,14 @@ public int getFracValue() {
         return fracPart.intValue();
     }
 
+    public void roundCeiling() {
+        value = value.setScale(0, 2);
+    }
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java:236]: use RoundingMode.ROUND_CEILING

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java [hunk 2]
```diff
@@ -231,6 +231,14 @@ public int getFracValue() {
         return fracPart.intValue();
     }
 
+    public void roundCeiling() {
+        value = value.setScale(0, 2);
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java:235]:
```suggestion
        value = value.setScale(0, RoundingMode.CEILING);
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java [hunk 3]
```diff
@@ -231,6 +231,14 @@ public int getFracValue() {
...
+    }
+
+    public void roundFloor() {
+        value = value.setScale(0, 3);
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java:239]:
```suggestion
        value = value.setScale(0, RoundingMode.FLOOR);
```

**Comment by yangzhg** [comment]
You should add some unit test to cover your 12 scenarios

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java (hunk 2 above)
Comment (xinyiZzz) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java]: Thats right

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java (hunk 3 above)
Comment (xinyiZzz) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DecimalLiteral.java]: Thats right

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**